test/OpenFeature.Contrib.Hooks.Otel.Test/MetricsHookTest.cs

```diff
@@ -0,0 +1,163 @@
+                .ConfigureResource(r => r.AddService("openfeature"))
+                .AddInMemoryExporter(exportedItems, option => option.PeriodicExportingMetricReaderOptions = new PeriodicExportingMetricReaderOptions { ExportIntervalMilliseconds = 100 })
+                .Build();
+
+            // Arrange
+            const string metricName = "feature_flag.evaluation_success_total";
+            var otelHook = new MetricsHook();
+            var evaluationContext = EvaluationContext.Empty;
+            var ctx = new HookContext<string>("my-flag", "foo", Constant.FlagValueType.String, new ClientMetadata("my-client", "1.0"), new Metadata("my-provider"), evaluationContext);
+
+            // Act
+            var hookTask = otelHook.After(ctx, new FlagEvaluationDetails<string>("my-flag", "foo", Constant.ErrorType.None, "STATIC", "default"), new Dictionary<string, object>());
+            // Wait for the metrics to be exported
+            Thread.Sleep(150);
+
+            // Assert
+            Assert.True(hookTask.IsCompleted);
+
+            // Assert metrics
+            Assert.NotEmpty(exportedItems);
+
+            // check if the metric is present in the exported items
+            var metric = exportedItems.FirstOrDefault(m => m.Name == metricName);
+            Assert.NotNull(metric);
+
+            var noOtherMetric = exportedItems.All(m => m.Name == metricName);
+            Assert.True(noOtherMetric);
+        }
+
+        [Fact]
+        public void Error_Test()
+        {
+            // Arrange metrics collector
+            var exportedItems = new List<Metric>();
+            Sdk.CreateMeterProviderBuilder()
+                .AddMeter("*")
+                .ConfigureResource(r => r.AddService("openfeature"))
+                .AddInMemoryExporter(exportedItems, option => option.PeriodicExportingMetricReaderOptions = new PeriodicExportingMetricReaderOptions { ExportIntervalMilliseconds = 100 })
+                .Build();
+
+            // Arrange
+            const string metricName = "feature_flag.evaluation_error_total";
+            var otelHook = new MetricsHook();
+            var evaluationContext = EvaluationContext.Empty;
+            var ctx = new HookContext<string>("my-flag", "foo", Constant.FlagValueType.String, new ClientMetadata("my-client", "1.0"), new Metadata("my-provider"), evaluationContext);
+
+            // Act
+            var hookTask = otelHook.Error(ctx, new Exception(), new Dictionary<string, object>());
+            // Wait for the metrics to be exported
+            Thread.Sleep(150);
+
+            // Assert
+            Assert.True(hookTask.IsCompleted);
+
+            // Assert metrics
+            Assert.NotEmpty(exportedItems);
+
+            // check if the metric is present in the exported items
+            var metric = exportedItems.FirstOrDefault(m => m.Name == metricName);
+            Assert.NotNull(metric);
+
+            var noOtherMetric = exportedItems.All(m => m.Name == metricName);
```

Comment:
Same [as above](https://github.com/open-feature/dotnet-sdk-contrib/pull/114/files#r1428084208), I think this can be added for each stage so we don't have attribute regressions.